### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by rcackerley**
@illiachaban Check out this implementation of promises

### CODE REVIEW

File: scripts/main.js [hunk 1]
```diff
@@ -120,21 +134,38 @@ var saveData = function(newOrder) {
...
+//           renderOrder(coffeeOrders[i], i);
+//         }
+//     }
+//   });
+// }
 
-}
+var loadData = function() {
+  var orderObjectToBeParsed
+  var loadedData = $.get('http://dc-coffeerun.herokuapp.com/api/coffeeorders')
```
Comment (illiaChaban) [inline scripts/main.js]: don't you need a new Promise here?

File: scripts/main.js [hunk 2]
```diff
@@ -28,7 +36,13 @@ var orderCompleted = function (event) {
     });
     console.log(orderToBeDeleted);
     orderToBeDeleted.classList.add('green-bg');
-    setTimeout(function() {orderToBeDeletedParent.removeChild(orderToBeDeleted)}, 2000);
+    var promise = new Promise(function(resolve) {
```
Comment (illiaChaban) [inline scripts/main.js]: not sure if it makes sense to use promise when you delete stuff, cause you don't have to wait in this case for any data

File: scripts/main.js (hunk 1 above)
Comment (rcackerley) [reply scripts/main.js]: $.get gives you a promise back so loadedData is now a promiseObject

File: scripts/main.js (hunk 2 above)
Comment (rcackerley) [reply scripts/main.js]: good point. thanks!

**Pull request merged**